### PR TITLE
Alert on Presence of Zero Logon Vulnerability - Slack

### PR DESCRIPTION
**Pull request by tharward-r7**
TODO:

- [ ] Test with actual VM trigger
- [ ] Successful Job screenshot

## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [x] Workflow verified as working in a test
- [x] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [x] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Workflow includes a detailed Description
- [x] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details (Slack Step?)
- [x] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [x] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [x] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

Post your results with screenshots in this PR using the following style:

### Screenshots

test

slack_output

workflow2

workflow1

### CODE REVIEW

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/workflow.spec.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  source_url: https://github.com/rapid7/insightconnect-workflows/tree/master/workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack
+  license_url: https://github.com/rapid7/insightconnect-workflows/blob/master/LICENSE
+  screenshots:
+    - name: slack_output.png
+      title: Zero Logon Vulnerability Detected Notification in Slack
+    - name: workflow1.png
+      title: Workflow Builder View
+    - name: workflow2.png
+      title: Workflow Builder View - Loop Step
+#    - name: job.png
```
Comment (cmchugh-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/workflow.spec.yaml:24]: should this be revised to reference the test.png, or will you be updating with a job png?

**Comment by cmchugh-r7** [comment]
looks good aside from needing the real job!

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/help.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+By default, the workflow detects specific vulnerabilities matching or containing the Zero Logon CVE: CVE-2020-1472. It can be easily updated for other pressing vulnerabilities by changing the ID within the "Is Zero Logon?" decision step within the "Loop Over found Vulnerabilities" loop. See InsightConnect's Help Documentation for more information on [using decision steps](https://docs.rapid7.com/insightconnect/lesson-4-use-decision-steps/#lesson-4-use-decision-steps) and on the [decision query language](https://docs.rapid7.com/insightconnect/format-query-language/#operators).
+ 
+## Technical Details
+
+Plugins utilized by workflow:
+
+|Plugin|Version|Count|
+|----|----|--------|
+|Rapid7 Vulnerability & Exploit Database|2.0.3|1|
+|Rapid7 InsightVM|4.7.0|1|
```
Comment (sengleson-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/help.md:45]:
```suggestion
```

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/help.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+# Version History
+
+* 1.0.0 - Initial workflow
+
+# Links
+
+## References
+
+* [Rapid7 Vulnerability & Exploit Database](https://rapid7.com/db)
+* [Slack](https://slack.com)
```
Comment (sengleson-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/help.md:61]:
```suggestion
* [Slack](https://slack.com)
* [InsightVM Events Trigger](https://docs.rapid7.com/insightconnect/set-up-an-insightvm-events-trigger)
```

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/help.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+Once the workflow has been imported, open the workflow and view the InsightVM Events trigger step. Save the trigger configuration step and proceed through the Instructions on subscribing to the InsightVM Vulnerabilities Found Webhook Event. This will require an [Insight Platform User API Key](https://docs.rapid7.com/insight/managing-platform-api-keys#generating-a-user-key). Insert your User API Key, copy the API request, run it from a command prompt, and look for a JSON response including an ID. If you receive a JSON response, then your webhook event subscription is active! Close your command prompt and return to the workflow in InsightConnect.
+
+After subscribing to the vulnerabilities found webhook event, return to the workflow editor and click the "Loop Over found Vulnerabilities" loop. Within the loop, click on the "Post to Slack" ChatOps step, and change the field "Enter #channel or @username" from `change_me` to a channel name in your Slack environment where you would like vulnerability notifications delivered. We recommend a private channel with representatives from security and remediation teams for the best visibility.
```
Comment (sengleson-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/help.md:30]:
```suggestion
After subscribing to the vulnerabilities found webhook event, return to the workflow editor and click the "Loop over Found Vulnerabilities" loop. Within the loop, click on the "Post to Slack" ChatOps step, and change the field "Enter #channel or @username" from `change_me` to a channel name in your Slack environment where you would like vulnerability notifications delivered. We recommend a private channel with representatives from security and remediation teams for the best visibility.
```

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/workflow.spec.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+name: Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack
+title: Alert on Presence of Zero Logon Vulnerability with Slack
+description: Receive Slack alerts for any Zero Logon (CVE-2020-1472 or related) vulnerability found
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [vulnerability_management, alerting_and_notifications]
+  keywords: [chat, slack]
```
Comment (sengleson-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/workflow.spec.yaml:12]:
```suggestion
  keywords: [chat, slack, cloud_enabled]
```

**Review by sengleson-r7 — CHANGES_REQUESTED** [review]

**Review by sengleson-r7 — DISMISSED** [review]

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/help.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
+# Description
+
+This workflow listens for InsightVM Webhook Events for newly discovered vulnerabilities, checks if a Zero Logon (CVE-2020-1472 or derivitave) vulnerability is present, and if found, posts a notification to a Slack channel.
```
Comment (jschipp-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/help.md:3]:
```suggestion
This workflow listens for InsightVM Webhook Events for newly discovered vulnerabilities, checks if a Zero Logon (CVE-2020-1472 or derivative) vulnerability is present, and if found, posts a notification to a Slack channel.
```

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack.icon [hunk 7]
```diff
@@ -0,0 +1,890 @@
...
+            "nodeId": "ecae1117-4c23-4a7f-af3b-0e53889d0212",
+            "name": "Post to Slack",
+            "type": "action_chatops",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "attachments": [],
+                "message": ":rotating_light: *ZERO LOGON VULNERABILITY DETECTED!* :rotating_light:\n:radioactive_sign: *Vulnerability Details:* :radioactive_sign:\n* *Title:* {{[e9e722f0-be7c-44dd-ac16-0262f632fef5].[content_result].[title]}}\n* *CVSS Score:* {{[e9e722f0-be7c-44dd-ac16-0262f632fef5].[content_result].[severity]}}\n* *More Details:* https://www.rapid7.com/db/vulnerabilities/{{[06bd742c-d3a6-4f68-856c-75fafd2a89a2].[$item]}}\n:computer: *Asset Details:* :computer:\n* *Hostname:* {{[79fddb0f-359c-4b94-98b1-cdd09e801247].[summary].[host_name]}}\n* *IP Address:* {{[79fddb0f-359c-4b94-98b1-cdd09e801247].[summary].[ip]}}\n* *Operating System:* {{[79fddb0f-359c-4b94-98b1-cdd09e801247].[summary].[operating_system]}}\n* *Last Scanned:* {{[79fddb0f-359c-4b94-98b1-cdd09e801247].[summary].[last_scanned]}}",
+                "recipient": "#change_me",
```
Comment (jschipp-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Slack.icon:560]:
```suggestion
                "recipient": "change_me",
```

**Review by jschipp-r7 — APPROVED** [review]

**Comment by jschipp-r7** [comment]
@tharward-r7 Can merge once retested

**Review by sengleson-r7 — APPROVED** [review]

**Pull request merged**